lms/djangoapps/verify_student/views.py

```diff
@@ -644,15 +652,14 @@ def _check_enrollment(self, user, course_key):
         return (has_paid, bool(is_active))
 
 
-def create_order_with_ecommerce_service(user, course_key, course_mode):     # pylint: disable=invalid-name
+def create_order_with_ecommerce_service(user, course_key, course_mode, processor):     # pylint: disable=invalid-name
     """ Create a new order using the E-Commerce API. """
     try:
         api = EcommerceAPI()
         # Make an API call to create the order and retrieve the results
-        _order_number, _order_status, data = api.create_order(user, course_mode.sku)
-
+        response_data = api.create_basket(user, course_mode.sku, processor)  # TODO get correct processor from request
         # Pass the payment parameters directly from the API response.
-        return HttpResponse(json.dumps(data['payment_parameters']), content_type='application/json')
+        return HttpResponse(json.dumps(response_data['payment_data'] or {}), content_type='application/json')
```

Comment:
I'm not sure returning `order` if `payment_data` is `None` is any better than returning an empty dict. Callers of this function expect payment data to be returned, so returning anything other than payment data, whether that's order data or an empty dict, will cause a failure later down the line, when the user tries to reach CyberSource. The following might even be fine:
```python
return HttpResponse(json.dumps(response_data.get('payment_data')), content_type='application/json')
```
If payment data is missing, this will cause a failure in [the JS responsible for constructing the form](https://github.com/edx/edx-platform/pull/7777/files#diff-fd4675021d17a3af130c8ded0c74a05cR126) POSTed to CyberSource.

What do you think? Ideally, we'd raise an exception and handle it gracefully.